fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewRule.java

```diff
@@ -383,30 +396,87 @@ protected Expression rewriteExpression(Expression sourceExpressionsToWrite, Plan
      * For another example as following:
      * predicate a = b in mv, and a = b and c = d in query, the compensatory predicate is c = d
      */
-    protected SplitPredicate predicatesCompensate(StructInfo queryStructInfo, StructInfo viewStructInfo,
-            SlotMapping queryToViewSlotMapping) {
+    protected SplitPredicate predicatesCompensate(
+            StructInfo queryStructInfo,
+            StructInfo viewStructInfo,
+            SlotMapping queryToViewSlotMapping,
+            ComparisonResult comparisonResult,
+            CascadesContext cascadesContext
+    ) {
+        // viewEquivalenceClass to query based
+        SlotMapping viewToQuerySlotMapping = queryToViewSlotMapping.inverse();
+        final Set<Expression> equalCompensateConjunctions = compensateEquivalence(
+                queryStructInfo,
+                viewStructInfo,
+                viewToQuerySlotMapping,
+                comparisonResult);
+        // range compensate
+        final Set<Expression> rangeCompensatePredicates = compensateRangePredicate(
+                queryStructInfo,
+                viewStructInfo,
+                viewToQuerySlotMapping,
+                comparisonResult);
+        // residual compensate
+        final Set<Expression> residualCompensatePredicates = compensateResidualPredicate(
+                queryStructInfo,
+                viewStructInfo,
+                viewToQuerySlotMapping,
+                comparisonResult);
+        // if the join type in query and mv plan is different, we should check and add filter on mv to make
+        // the mv join type is accord with query
+        Set<Set<Slot>> viewNoNullableSlot = comparisonResult.getViewNoNullableSlot();
```

Comment:
maybe `requireNoNullableSlot ` name is better

Reply:
OK, it's better